File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length-1;

        int mid = (left+right)/2;

        for(int i = 0; i<nums.length; i++){
            if(nums[i]==target){
                return i;
            }

            else if(nums[i]<target){
                right = mid-1;
            }

            else{
                left = mid+1;
            }
        }

        return -1;
    }
}

File: 792-binary-search/binary-search.cpp
class Solution {
public:
    int binarySearch(vector<int>& nums, int l, int h, int target){
        while(l<=h){
            int mid = l + (h-l)/2;

            if(nums[mid] == target){
                return mid;
            }else if (nums[mid]>target){
                return binarySearch(nums, l, mid-1, target);
            }else{
                return binarySearch(nums, mid+1, h, target);
            }
        }
        
        return -1;
    }

    int search(vector<int>& nums, int target) {
        int l = 0, h = nums.size() - 1;

        return binarySearch(nums, l, h, target);
    }
};